src/coreclr/CMakeLists.txt

```diff
@@ -129,6 +129,8 @@ endif(CLR_CROSS_COMPONENTS_BUILD)
 #-------------------
 include(pgosupport.cmake)
 
+add_subdirectory(src/libraries-native)
```

Comment:
We can define `${CLR_REPO_ROOT_DIR}` just like we are using `${CLR_ENG_NATIVE_DIR}` on line 118, it is passed from:

https://github.com/dotnet/runtime/blob/932baca94e88ac77a5a70b2032d3341603b4610e/eng/native/build-commons.sh#L114-L117

```diff
- __CMakeArgs="-DCLR_ENG_NATIVE_DIR=\"$engNativeDir\" $__CMakeArgs" 
+ __CMakeArgs="-DCLR_ENG_NATIVE_DIR=\"$engNativeDir\" $__CMakeArgs -DCLR_REPO_ROOT_DIR=\"$__RepoRootDir\" $__CMakeArgs"
```

same thing at: https://github.com/dotnet/runtime/blob/932baca94e88ac77a5a70b2032d3341603b4610e/src/coreclr/build.cmd#L460

after that we can replace this line with:

```cmake
set(CMAKE_INCLUDE_CURRENT_DIR ON)
set(CORECLR_BUILD ON)

if(CLR_CMAKE_HOST_UNIX AND NOT DEFINED CLR_CROSS_COMPONENTS_BUILD)
    add_subdirectory(${CLR_REPO_ROOT_DIR}/src/libraries/Native/Unix/System.Globalization.Native System.Globalization.Native)
endif()
```

without a separate `src/coreclr/src/libraries-native/CMakeLists.txt` proxy file.

Reply:
I’d still rather have it in a separate CMakeLists file because we don’t want to leak the CMAKE_INCLUDE_CURRENT_DIR setting or the other include directories.

Reply:
one pattern which is used in certain places for that purpose is:

```cmake
set(SOMETHING ON)

# do stuff

set(SOMETHING) # without args, to unset
```

Reply:
The proxy file was intended because in the future we might link more of the shims in order to be able to do runtime calls, so then we would just include and add subdirectories here without necessarily complicating `src/coreclr/CMakeLists.txt` -- also that way we keep the set and defines isolated for libraries-native, I like the separation as it is cleaner.

> We can define ${CLR_REPO_ROOT_DIR}

I thought about that, but talking to @jkoritzinsky he recommended to avoid passing more defines via command line as it is nice to keep the cmake command simple so that you basically can just do cmake and it builds fine without having to learn all the parameters.

Reply:
@janvorli recommended to add `CLR_REPO_ROOT_DIR` instead of hardcoding relative paths in my previous PR. There are about dozen of arguments passed to the final cmake call (and counting), so i guess it would be aligned with where we are heading.

Reply:
I would also prefer not to use the long relative paths that are hard to read. Defining `CLR_REPO_ROOT_DIR` would be my preference. As @am11 says, we already have quite some defines that are necessary.

Reply:
Ok sounds good, I'll add that as well.